packages/gestalt/src/TextArea.js

```diff
@@ -34,137 +35,118 @@ type Props = {|
+    setFocused(false);
     if (onBlur) {
       onBlur({ event, value: event.currentTarget.value });
     }
   };
 
-  handleFocus: (
-    event: SyntheticFocusEvent<HTMLTextAreaElement>
-  ) => void = event => {
-    const { onFocus } = this.props;
+  const handleFocus = (event: SyntheticFocusEvent<HTMLTextAreaElement>) => {
+    setFocused(false);
     if (onFocus) {
       onFocus({ event, value: event.currentTarget.value });
     }
   };
 
-  handleKeyDown: (
+  const handleKeyDown = (
     event: SyntheticKeyboardEvent<HTMLTextAreaElement>
-  ) => void = event => {
-    const { onKeyDown } = this.props;
+  ) => {
     if (onKeyDown) {
       onKeyDown({ event, value: event.currentTarget.value });
     }
   };
 
-  render(): React.Node {
-    const {
-      disabled,
-      errorMessage,
-      hasError,
-      helperText,
-      id,
-      label,
-      name,
-      placeholder,
-      rows,
-      value,
-    } = this.props;
+  const classes = classnames(
+    styles.textArea,
+    formElement.base,
+    disabled ? formElement.disabled : formElement.enabled,
+    hasError || errorMessage ? formElement.errored : formElement.normal
+  );
+
+  return (
+    <span>
+      {label && <FormLabel id={id} label={label} />}
+      <textarea
+        aria-describedby={errorMessage && focused ? `${id}-error` : null}
+        aria-invalid={errorMessage || hasError ? 'true' : 'false'}
+        className={classes}
+        disabled={disabled}
+        id={id}
+        name={name}
+        onBlur={handleBlur}
+        onChange={handleChange}
+        onFocus={handleFocus}
+        onKeyDown={handleKeyDown}
+        placeholder={placeholder}
+        ref={forwardedRef}
+        rows={rows}
+        value={value}
+      />
+      {helperText && !errorMessage ? (
+        <FormHelperText text={helperText} />
+      ) : null}
+      {errorMessage && <FormErrorMessage id={id} text={errorMessage} />}
+    </span>
+  );
+}
 
-    const { focused } = this.state;
+TextArea.propTypes = {
+  disabled: PropTypes.bool,
+  errorMessage: PropTypes.string,
+  forwardedRef: PropTypes.oneOfType([
+    PropTypes.func,
+    PropTypes.shape({
+      current: PropTypes.any,
+    }),
+  ]),
+  hasError: PropTypes.bool,
+  helperText: PropTypes.string,
+  id: PropTypes.string.isRequired,
+  name: PropTypes.string,
+  label: PropTypes.string,
+  onBlur: PropTypes.func,
+  onChange: PropTypes.func.isRequired,
+  onFocus: PropTypes.func,
+  onKeyDown: PropTypes.func,
+  placeholder: PropTypes.string,
+  rows: PropTypes.number,
+  value: PropTypes.string,
+};
+
+function forwardRef(props, ref) {
+  return <TextArea {...props} forwardedRef={ref} />;
+}
 
-    const classes = classnames(
-      styles.textArea,
-      formElement.base,
-      disabled ? formElement.disabled : formElement.enabled,
-      hasError || errorMessage ? formElement.errored : formElement.normal
-    );
+forwardRef.displayName = 'TextArea';
```

Comment:
@AlbertCarreras see https://github.com/pinterest/gestalt/blob/a4c967d97d9fc93613ea4e6442c66b9c64262ca2/packages/gestalt/src/Link.js#L154-L163

we should be setting the `displayName` on the default export as well.

I recommend the following:
```
function TextAreaWithRef(props, ref) {
  return <TextArea {...props} forwardedRef={ref} />;
}
TextAreaWithRef.displayName = 'ForwardRef(TextArea)';

const TextAreaWithForwardRef: React.AbstractComponent<
  Props,
  HTMLAnchorElement
> = React.forwardRef<Props, HTMLAnchorElement>(LinkWithRef);

TextAreaWithForwardRef.displayName = 'TextArea';

export default TextAreaWithForwardRef;
```

Reply:
done